tools/lint.toml

```diff
@@ -0,0 +1,24 @@
+[tool.ruff]
+line-length = 88
+
+# Enable Pyflakes `E` and `F` codes by default.
+select = ["E", "F"]
+ignore = [
+  "E121", "E122", "E123", "E125", "E126", "E127", "E128", "E226",
+  "E251", "E265", "E266", "E302", "E402", "E712", "E721", "E731",
+  "E741", "W291", "W293", "W391", "W503", "W504"
+]
+excludes = [
+  'scipy/datasets/_registry.py'
+]
+per-file-ignores = {'**/__init.py', ['F401', 'F403']}
+
+# Allow unused variables when underscore-prefixed.
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
+
+# Assume Python 3.10.
+target-version = "py38"
```

Comment:
Should we update the `target-version` to match the comment/newer Python above?

One more question about this file--where is `lint.toml` used? Is it just that you can use `ruff` with any `.toml` file at appropriate location?

A quick browse of their GitHub page used other filenames so wasn't sure:
> As an alternative to pyproject.toml, Ruff will also respect a ruff.toml file, which implements an equivalent schema (though the [tool.ruff] hierarchy can be omitted)

Reply:
Thanks, Tyler. `lint.py` explicitly sets `--config=lint.toml`.

Reply:
Recommended entrypoints for the linter are: `python dev.py lint` or `tools/lint.py`.

I think if we wanted it to pick up `ruff.toml` by default, we'd have to move it to the root of the repo, or include its config in our `pyproject.toml`.

Reply:
Ah, I didn't catch that in the diff--just to confirm, this is separate from the usage of `lint.ini` in `tools/lint.py`? Or does it remap to to the `toml` file from that?

Reply:
Scratch all my comments: `ruff` expects `ruff.toml` (even if `--config` is used), and `lint.py` didn't specify it correctly.

Reply:
Unfortunately we do need the file to _either_ be named `ruff.toml` or `pyproject.toml`. Otherwise, we have no way of knowing which schema to enforce. (The schemas are the ~same between `ruff.toml` and `pyproject.toml`, except that all the Ruff settings are nested under `[tool.ruff]`.)

Reply:
Thanks @charliermarsh. It sounds like you may need two config options then, `--config` and `--config-pyproject` or similar. I think it's quite confusing not to honor the `--config` option just because the file is named incorrectly.

Reply:
As an alternative, if the file _isn't_ named `pyproject.toml`, I could just treat it as in `ruff.toml` format. That would at least be an improvement, I think, though would still have the potential to cause confusion as you describe.

Reply:
@stefanv - This is changed in the next release (anything not named `pyproject.toml` is assumed to be in `ruff.toml` format).